### PR TITLE
Add covariance serialization

### PR DESCRIPTION
**Pull request by QRemy**
- add Covariance.read() and modify Covariance.write() to work with models serialization
- remove .to_table() which is broken if parameters have duplicate names.
  For now unique names are generated on .write() using make_name() but once 
  models won't have duplicate parameters names, we could use param_name +
 model_name that will make the covariance matrix files more readable.

### CODE REVIEW

File: gammapy/modeling/models/core.py [hunk 1]
```diff
@@ -274,7 +282,13 @@ def to_dict(self):
         for model in self._models:
             model_data = model.to_dict()
             models_data.append(model_data)
-        return {"components": models_data}
+        if self._covariance.filename is not None:
+            return {
+                "components": models_data,
+                "covariance": str(self._covariance.filename),
```
Comment (adonath) [inline gammapy/modeling/models/core.py]: I think here we can just rely on a default filename create from the models filename like `models.yaml` and `models_covariance.ecsv` or `my_model.yaml` and `my_model_covariance.ecsv`. More flexibility is then provided with the `Model.write_covariance()` and `Model.read_covariance()` API.

**Comment by adonath** [comment]
Just to add: the `Covariance.read()` and `Covariance.write()` can be removed then...

File: gammapy/modeling/models/core.py (hunk 1 above)
Comment (QRemy) [reply gammapy/modeling/models/core.py]: I define the name on models.write() using the same prefix than for the models.yaml file, now it is passed  through self._covar_file

Comment (adonath) [reply gammapy/modeling/models/core.py]: OK, sounds good...

**Pull request merged**